src/iccpd/src/iccp_csm.c

```diff
@@ -466,7 +490,11 @@ int iccp_csm_prepare_capability_msg(struct CSM* csm, char* buf, size_t max_buf_s
     LDPICCPCapabilityTLV* cap = (LDPICCPCapabilityTLV*)&buf[sizeof(LDPHdr)];
     size_t msg_len = sizeof(LDPHdr) + sizeof(LDPICCPCapabilityTLV);
 
+#ifdef __STDC_LIB_EXT1__
+    memset_s(buf, sizeof(*buf), 0, max_buf_size);
```

Comment:
>buf [](http://example.com/codeflow?start=26&length=3)

This is a special case, since buf is a pointer and you do not know the buf size for sure. Suggest `memset_s(buf, max_buf_size, 0, max_buf_size);`